packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port_utils.py

```diff
@@ -68,16 +68,45 @@ async def get_value_from_link(
 
 async def get_download_link_from_storage(
     user_id: int, value: FileLink, link_type: LinkType
-) -> Optional[AnyUrl]:
+) -> AnyUrl:
+    """
+    :raises exceptions.NodeportsException
+    :raises exceptions.S3InvalidPathError
+    :raises exceptions.StorageInvalidCall
+    :raises exceptions.StorageServerIssue
+    """
     log.debug("getting link to file from storage %s", value)
+
     link = await filemanager.get_download_link_from_s3(
         user_id=user_id,
         store_id=f"{value.store}",
         store_name=None,
         s3_object=value.path,
         link_type=link_type,
     )
-    return parse_obj_as(AnyUrl, f"{link}") if link else None
+
+    # could raise ValidationError but will never do it since
+    assert isinstance(link, URL)  # nosec
+    return parse_obj_as(AnyUrl, f"{link}")
+
+
+async def get_download_link_from_storage_v2(
```

Comment:
haha no,don't worry, there is NOT a ``storage_v2`` . It is to distinguish it from the other since is the same function with different arguments.
Please read the comment below the function. It should explain the situation ... otherwise i can clarify further

Reply:
Maybe append `overload` instead of `v2`. Might be more useful for the users. Yes I failed to read the name of the function in the comment.
Also in the comment you say `get_upload_link_from_storage` but the name of the function is `get_download_link_from_storage_v2`.